### ISSUE TITLE
ModuleRef build issue

### ISSUE THREAD

**Issue by goncalves-diogo** [nestjs/nest#7877]
## Bug Report
Hi, I'm currently experiencing a bug regarding the ModuleRef from nestjs/core, I'm unsure if this the right place to ask or not.
## Current behavior

Currently I'm trying to create a library where i use the ModuleRef to create a selector. For example, given a dictionary of containing the name of class, i use the ModuleRef to actually get the class. This works as expected when directly import the code. However since this is meant to be used across diferent microservices i want to use it as NPM package. In order to do so i created a new repository with that code. So far so good. Now when compiling the code to use i use 
```
npm run build 
```
 which itself uses tsc to compile with the current options
```
{
    "compilerOptions": {
        "module": "commonjs",
        "declaration": true,
        "removeComments": true,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "allowSyntheticDefaultImports": true,
        "target": "ES6",
        "sourceMap": true,
        "outDir": "./dist",
        "baseUrl": "./",
        "incremental": true
    },
    "exclude": [
        "node_modules",
        "test",
        "**/*spec.ts"
    ]
}

```
Now, it will generate the dist/index.js.
At this point I export it as NPM package (I've build 5 other application with this process which works so the problem isn't up until now). 

When i try to import the package via 
```
import { Selector } from '@beevo/ma-libs-selector'
```
The program stops working complaining about ModuleRef not being imported correctly.

My humble guess at this point is that ModuleRef being build to js or even ts just won't work, Is there any work around to this, or this isn't an issue at all.
## Input Code

```ts
const your = (code) => here;
```

## Possible Solution

For the code being exported to work the same way as the code built 

## Environment

<pre><code>
Nest version: 7.5.3

 
For Tooling issues:
- Node version: 14.15.3  
- Platform:  Linux, Ubuntu 

Others:

Sorry i couldn't provide any real code, as this is a private repository and i'm not sure what i can share, If need be i will share code
</code></pre>

**Comment by jmcdo29** [comment]
Please provide a [minimum reproduction repository](http://minimum-reproduction.wtf).

**Comment by kamilmysliwiec** [comment]
> My humble guess at this point is that ModuleRef being build to js or even ts just won't work, Is there any work around to this, or this isn't an issue at all.

`@nestjs/core` and `@nestjs/common` packages should be defined as **peer** dependencies of your library

**Comment by goncalves-diogo** [comment]
Sorry my issue wasn't well explained, I've tried the peerDepencies and it still doesn't work. 

In order to help with the bug i've created a repo to share my code, it would be amazing if you take a look into it. I shared the library not the usage file itself, if need be i will also share how i use this code.
Thanks in advance

https://github.com/goncalves-diogo/mbr_nestjs

**Comment by kamilmysliwiec** [comment]
Not sure if I understand. How can we reproduce this issue with just a library?

Also, https://github.com/goncalves-diogo/mbr_nestjs/blob/master/src/selector.service.ts#L32 this won't work properly unless all target classes are registered as providers within the same module as the `SelectorService` as `ModuleRef`, when used within providers, runs in the `strict` mode by default (you can read about this in the docs).

**Comment by goncalves-diogo** [comment]
So, I've figure out the bug in my code, which in the end was something really stupid, I'm not really familiar with package.json configuration or even Nestjs itself.

That being said I don't really understand what happened, or if this is expected behavior or not.

To contextualize in my companies projects I've installed all my dependencies with the following format.
```
    "dependencies": {
        "@mycompanyname/mypackage": "bitbucket-URL",
    },
```

Something like the previous example always worked properly. However when developing this specific feature this way of importing apparently didn't work. I changed the previous import to the following 
```
    "dependencies": {
        "mypackage": "bitbucket-URL",

    },
```
And Magically it started working. I'm just baffled how that starting working...

I think i wasn't able to express myself correctly. So thank you so much for your time.
